projects/kit/src/lib/utils/date/tests/append-date.spec.ts

```diff
@@ -179,11 +169,92 @@ describe('appendDate', () => {
         expect(result.getDate()).toBe(31);
     });
 
-    it('year: 2025, month: 1, day: 27, if {month: 1} was passed', () => {
-        const result = appendDate(new Date(2025, 0, 31), {month: 1});
+    it(
+        'year: 2025, month: 8, day: 1, if {month: -1} was passed (for last ' +
+            'day of month, when current month has less days, then final month)',
+        () => {
+            const result = appendDate(new Date(2025, 8, 30), {month: -1});
+
+            expect(result.getFullYear()).toBe(2025);
+            expect(result.getMonth()).toBe(8);
+            expect(result.getDate()).toBe(1);
+        },
+    );
+
+    it('year: 2025, month: 8, day: 30, if {month: 1} was passed', () => {
+        const result = appendDate(new Date(2025, 8, 1), {month: 1});
 
         expect(result.getFullYear()).toBe(2025);
-        expect(result.getMonth()).toBe(1);
-        expect(result.getDate()).toBe(27);
+        expect(result.getMonth()).toBe(8);
+        expect(result.getDate()).toBe(30);
+    });
+
+    it(
+        'year: 2018, month: 2, day: 1, if {month: -1} was passed (for last day' +
+            ' of month, when current month has more days, then final month)',
+        () => {
+            const result = appendDate(new Date(2018, 2, 31), {month: -1});
+
+            expect(result.getFullYear()).toBe(2018);
+            expect(result.getMonth()).toBe(2);
+            expect(result.getDate()).toBe(1);
+        },
+    );
+
+    it(
+        'year: 2018, month: 1, day: 27, if {month: -1} was passed (when current' +
+            ' month has more days, then final month, and final month don`t has the day)',
+        () => {
+            const result = appendDate(new Date(2018, 2, 30), {month: -1});
+
+            expect(result.getFullYear()).toBe(2018);
+            expect(result.getMonth()).toBe(1);
+            expect(result.getDate()).toBe(28);
+        },
+    );
+
+    it(
+        'year: 2018, month: 1, day: 26, if {month: -1} was passed (when current' +
+            ' month has more days, then final month, and final month don`t has the day)',
+        () => {
+            const result = appendDate(new Date(2018, 2, 29), {month: -1});
+
+            expect(result.getFullYear()).toBe(2018);
+            expect(result.getMonth()).toBe(1);
+            expect(result.getDate()).toBe(27);
+        },
+    );
+
+    it(
+        'year: 2018, month: 1, day: 26, if {month: -1} was passed (when current' +
+            'month has more days, then final month, but both has the day, and' +
+            ' it`s last day of final month)',
+        () => {
+            const result = appendDate(new Date(2018, 2, 28), {month: -1});
+
+            expect(result.getFullYear()).toBe(2018);
+            expect(result.getMonth()).toBe(2);
+            expect(result.getDate()).toBe(1);
```

Comment:
Is it correct ? 🤔 

28th March is not the last day of month.
1st March + {month: 1} = last day of March, is not it ?